### ISSUE TITLE
[Outreachy Task Submission] Accessibility issue with Share modal form labels

### ISSUE THREAD

**Issue by Chiemezuo** [ushahidi/platform#4878]
# Intro
Accessibility-checking software flags the `share` modal when it opens after clicking on the `share` button at the top of every page because there is no appropriate label for the search field. This fix addresses the problem by modifying the code from using `div` to using `label`, for better semantic value, especially because the `div` there is only used for styling.

## How to reproduce this issue.
If you use a tool like [WAVE evaluation tool](https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh), you can follow these steps:

1. Open your browser and log into your running Ushahidi deployment (from localhost).
2. Click on the `Share` button at the top part of the screen. (See picture below for confirmation)

![image](https://github.com/ushahidi/platform/assets/29470516/7968b5fe-adcd-4c60-a210-34d9d687993f)

3. When the `Share` modal opens, put on the Accessibility browser extension. It will flag the form input fields.

## Why this issue happens
It happens because despite there being a form label in the code, the field and the label are not linked with `for` and `id` attributes. See snippet below:
```html
      <mat-label>{{ 'share.web_address' | translate }}</mat-label>
      <mat-form-field appearance="outline">
        <input
          matInput
          [(ngModel)]="address"
          (input)="changeAddress($event)"
          [data-qa]="'survey-web-address'"
        />
      </mat-form-field>
```
Angular's `<mat-label>` element doesn't quite support the `for` attribute

## Accessibility Violation
Success Criterion 3.3.2 Labels or Instructions

 (Level A)
[Labels](https://www.w3.org/TR/WCAG22/#dfn-labels) or instructions are provided when content requires user input.
